src/test/ui/consts/const-eval/const_assume.rs

```diff
@@ -0,0 +1,17 @@
+// check-pass
+
+// Check that `const_assume` feature allow `assume` intrinsic
+// to be used in const contexts.
+
+#![feature(core_intrinsics, const_assume)]
+
+extern crate core;
+
+use core::intrinsics::assume;
+
+pub const unsafe fn foo(x: usize, y: usize) -> usize {
+    assume(y != 0);
+    x / y
+}
```

Comment:
Isn't this the kind of test that should be made a unit test instead (https://github.com/rust-lang/rust/issues/76268)?

Reply:
I only saw `const_slice_from_raw_parts` and `const_raw_ptr_deref` in `library/core/tests/lib.rs`.
Don't really sure about this.

Reply:
Yeah, I think this should be a unit-test.

Reply:
It's kind of weird since most intrinsic not tested in `library/core/tests`.
Any, I moved it.